docs/scripts/codecs.sh

```diff
@@ -0,0 +1,10 @@
+for f in $(find ../src/codec/*.rs | sed -e "s;../src/codec/;;g" -e 's;.rs$;;')
+do
+  name=$(echo ${f} | sed -e 's/_/-/g')
+  file="../src/codec/${f}.rs"
+  doc_name=$(echo ${f} | sed -e 's;^.*/;;g')
+  doc_file="codecs/${doc_name}.md"
+  echo "${doc_file}"
+  echo "# ${name}\n" > "${doc_file}"
+  cat "${file}" | grep '^//!' | sed -e 's;^//! ;;g' -e 's;^//!;;g' >> "${doc_file}"
```

Comment:
can't `rustdoc` generate some md for us?

Reply:
not really sadly, that was my first attempt but rustdoc doesn't allow filtering to what it generates to it'll just generate the whole and extracting the relevant part from a `md` is harder than from a `rs` as it lacks the structure we have in the rust file.